### ISSUE TITLE
Stage a update to job definition but don't trigger a deployment

### ISSUE THREAD

**Issue by thefallentree** [hashicorp/nomad#18877]
### Proposal

Allow "staging" a job definition update , but don't trigger a deployment.

### Use-cases

Some time, we want to be able to upgrade the job definition to a new version, but don't want to interrupt the current running job yet,  it can be picked up upon a restart of task/allocation. 
 

### Attempted Solutions
N/A

**Comment by tgross** [comment]
Hi @thefallentree! Thanks for this feature suggestion.

At first glance this looks easy to implement, as we'd simply not write a new evaluation as part of the `Job.Register` RPC handler. However, the very first evaluation triggered for that job after we did that would result in a new Deployment getting created, as it's the scheduler that creates the deployment object and not the initial job submission. So this would require some new state to get written to the Job that would somehow get toggled, and then new logic in the scheduler to handle that case.

It's an interesting idea and I'll mark it for further discussion and possible roadmapping. It might help bump it in priority if you could describe some specific business use cases you have for this sort of feature.

**Comment by thefallentree** [comment]
certainly:  as a real business case: we found a bug in our deployed software, that will make it crash upon restart.  But we also have a maintainece window that we want to prefer upgrade the deployed software at that window to minimize customer disruption.

But given the emphermal nature of cloud environment,  any machine can fail at any time , if a server failed, we do want it to come back up with the new version instead of the old version. Hence that we want to stage a update to the job definition so when they come back-up, they can come up correctly.
